### PR TITLE
fix(datasets): unpin pandas-gbq

### PR DESCRIPTION
**Pull request by felipemonroy**
## Description
More recent versions of pandas does not work with `pandas-gbq<0.18.0`, which was set for `python < 3.11` 

## Development notes
Not sure why the restriction was there

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by felipemonroy** [comment]
@astrojuanlu @noklam could you review it please?

**Comment by noklam** [comment]
https://github.com/kedro-org/kedro/pull/2068

It was a while ago and we haven't updated the pin since then. Thanks for raising the PR.

File: kedro-datasets/pyproject.toml [hunk 1]
```diff
@@ -89,8 +89,8 @@ pandas-csvdataset = ["kedro-datasets[pandas-base]"]
 pandas-deltatabledataset = ["kedro-datasets[pandas-base]", "deltalake>=0.10.0"]
 pandas-exceldataset = ["kedro-datasets[pandas-base]", "openpyxl>=3.0.6, <4.0"]
 pandas-featherdataset = ["kedro-datasets[pandas-base]"]
-pandas-gbqtabledataset = ["kedro-datasets[pandas-base]", "pandas-gbq>=0.12.0, <0.18.0; python_version < '3.11'", "pandas-gbq>=0.18.0; python_version >= '3.11'"]
-pandas-gbqquerydataset = ["kedro-datasets[pandas-base]", "pandas-gbq>=0.12.0, <0.18.0; python_version < '3.11'", "pandas-gbq>=0.18.0; python_version >= '3.11'"]
+pandas-gbqtabledataset = ["kedro-datasets[pandas-base]", "pandas-gbq>=0.18.0"]
+pandas-gbqquerydataset = ["kedro-datasets[pandas-base]", "pandas-gbq>=0.18.0"]
```
Comment (noklam) [inline kedro-datasets/pyproject.toml:93]:
```suggestion
pandas-gbqtabledataset = ["kedro-datasets[pandas-base]", "pandas-gbq>=0.12.0"]
pandas-gbqquerydataset = ["kedro-datasets[pandas-base]", "pandas-gbq>=0.12.0"]
```

Comment (noklam) [reply kedro-datasets/pyproject.toml:93]: I don't think we need to bump the lower pin here, removing the upper pin is enough.

**Review by astrojuanlu — APPROVED** [review]
Thanks @felipemonroy !

**Comment by astrojuanlu** [comment]
Hey @felipemonroy , do you still intend to pursue this PR?

**Comment by felipemonroy** [comment]
Hi @astrojuanlu. After rebase my tests were failing because of #755 . I had to make #773 because of that. My git flow was a mess so I forked the repo again (and lost all the previous changes). I am going to create the PR again tomorrow.

**Pull request closed without merging**